api/internal/localizer/localizer.go

```diff
@@ -127,3 +135,66 @@ func (lc *Localizer) localizeFile(path string) (string, error) {
 	}
 	return locPath, nil
 }
+
+// localizeBuiltinPlugins localizes built-in plugins on kust that can contain file paths. The built-in plugins
+// can be inline or in a file. This excludes helm.
+//
+// Note that the localization in this function has not been implemented yet.
+func (lc *Localizer) localizeBuiltinPlugins(kust *types.Kustomization) error {
+	for fieldName, plugins := range map[string]struct {
+		entries   []string
+		localizer kio.Filter
+	}{
+		"generators": {
+			kust.Generators,
+			&localizeBuiltinGenerators{},
+		},
+		"transformers": {
+			kust.Transformers,
+			&localizeBuiltinTransformers{},
+		},
+	} {
+		for i, entry := range plugins.entries {
+			rm, isPath, err := lc.loadResource(entry)
+			if err != nil {
+				return errors.WrapPrefixf(err, "unable to load %s entry", fieldName)
+			}
+			err = rm.ApplyFilter(plugins.localizer)
+			if err != nil {
+				return errors.Wrap(err)
+			}
+			localizedPlugin, err := rm.AsYaml()
+			if err != nil {
+				return errors.WrapPrefixf(err, "unable to serialize localized %s entry %q", fieldName, entry)
+			}
+			var newEntry string
+			if isPath {
+				// TODO(annasong): write localizedPlugin to dst
+				newEntry = entry
+			} else {
+				newEntry = string(localizedPlugin)
+			}
+			plugins.entries[i] = newEntry
+		}
+	}
+	if len(kust.Validators) > 0 {
+		return errors.Errorf("no built-in validators with file paths, but validators field non-empty")
```

Comment:
Could you explain why we need to throw an error here? Are we always throwing an error if we call `localize` on a kustomization file with a validators field?

Reply:
Are you saying that if I have a kustomization file that has

```
validators:
- {some custom validator without any field paths}
```

and I run `kustomize localize` on it, it would return an error?

I don't think that's correct behavior. IMO `kustomize localize` should preserve the validators field as it is, the same way we treat other fields that don't have file paths.

Reply:
This would introduce an inconsistency. We only planned to handle `generators` and `transformers` with file paths and throw an error for all the others. Do we want to change the behavior for all these fields?

Reply:
`validators` doesn't accept `PatchStrategicMergeTransformer` that is identical to the original resource. However, `ReplacementTransformer` works, so I enabled `validator` localization in my latest commit.

Ready for another look!